src/utils/Dimensions.ts

```diff
@@ -0,0 +1,33 @@
+module n3Charts.Utils {
+  'use strict';
+
+  export class Dimensions {
+    public width = 600;
+    public height = 200;
+    public innerWidth = 560;
+    public innerHeight = 160;
+    public margin = {
+      left: 30,
+      bottom: 20,
+      right: 20,
+      top: 20
+    };
+
+    getPixelCssProp(element: HTMLElement, propertyName: string) {
+      var style = window.getComputedStyle(element, null);
+      return +style.getPropertyValue(propertyName).replace(/px$/, '');
+    }
+
+    fromParentElement(parent: HTMLElement) {
+      // Oooooh I hate doing this.
```

Comment:
sure, box sizing is used for the width calculation, and with `border-box` padding is not increasing the width. In the demo you can see, that only the element on the right has width and height of 100px.
Here are the docs: https://developer.mozilla.org/en/docs/Web/CSS/box-sizing
Here is a demo: http://plnkr.co/edit/RfmnhtQUwOaFkCDFHS15?p=preview